src/Controls/StreetControls.js

```diff
@@ -77,8 +79,143 @@ class StreetControls extends FirstPersonControls {
+                this.hideSurfaces();
+                return;
+            }
+
+            // get distance from the elevation layer
+            const distanceToGround = this.camera.position.distanceTo(target);
+            // compare first picked object distance to the distance of the elevation layer
+            if (!firstPicked.object.isTileMesh && firstPicked.distance < distanceToGround) {
+                // here we pick something, for example a building
+                target.copy(firstPicked.point);
+                // update recangle surface helper
+                this.updateSurface(target, firstPicked.face.normal, this._rectangleMesh, this._circleMesh);
+                // callback
+                pickObject(target);
+            } else {
+                // here we pick the ground, update circle surface helper
+                this.updateSurface(target, firstPicked.face.normal, this._circleMesh, this._rectangleMesh);
+                // callback
+                pickGround(target);
+            }
+        }
+    }
+
+    onClick() {
+        // here we detect mouse click when mouse is not moving,
+        // to avoid interfering with mouse drag movement, managed by FirstPersonControls
+        this.maybeClickWithoutMoving = true;
+    }
+
+    onClickWithoutMoving(event) {
+        this.pick(event, this.onClickOnGround.bind(this), this.onClickOnWall.bind(this));
+    }
+
+    onMouseUp(e) {
+        super.onMouseUp();
+
+        if (this.maybeClickWithoutMoving) {
+            this.onClickWithoutMoving(e);
+        }
+    }
+
+    onMouseMove(event) {
+        super.onMouseMove(event);
+
+        if (!this._isMouseDown) {
+            this.pick(event);
+        } else {
+            this.maybeClickWithoutMoving = false;
+        }
+    }
+
+    // create surfaces, a circle and a rectangle used to help end user
+    initSurfaces() {
+        const material = new THREE.MeshBasicMaterial({
+            side: THREE.DoubleSide,
+            color: 0xffffff,
+            depthTest: false,
+            transparent: true,
+            opacity: 0.5 });
+
+        const geomPlane = new THREE.PlaneGeometry(4, 2, 1);
+        this._rectangleMesh = new THREE.Mesh(geomPlane, material);
+        const geomCircle = new THREE.CircleGeometry(1, 32);
+        this._circleMesh = new THREE.Mesh(geomCircle, material);
+
+        this.view.scene.add(this._rectangleMesh);
+        this.view.scene.add(this._circleMesh);
+
+        this._rectangleMesh.visible = false;
+        this._circleMesh.visible = false;
+
+        this._rectangleMesh.rotateX(Math.PI / 2);
+
+        // surface node is used to store the surfaces
+        this.surfacesNode = new THREE.Object3D();
+        this.view.scene.add(this.surfacesNode);
+
+        this.surfacesNode.add(this._rectangleMesh);
+        this.surfacesNode.add(this._circleMesh);
+    }
+
+    // hide both rectangle and circle helpers
+    hideSurfaces() {
+        this._circleMesh.visible = false;
```

Comment:
Could you use a `THREE.Group` to manage the `surfaceHelpers`. You could write only `surfaceHelpers.visible = false`.

Reply:
No because, on the other hand, when I will want to make only one surface visible, I'll have to make the Group visible again, so this idea remove on line of code somewhere, but make me add another line at other place. So I think it's not a good idea.